packages/flutter/test/material/range_slider_test.dart

```diff
@@ -0,0 +1,1214 @@
+    final Offset bottomRight = tester.getBottomRight(find.byType(RangeSlider)).translate(-24, 0);
+    final Offset middle = topLeft + bottomRight / 2;
+
+    // Drag the start thumb towards the center.
+    final Offset leftTarget = topLeft + (bottomRight - topLeft) * 0.3;
+    await tester.dragFrom(leftTarget, middle - leftTarget);
+    expect(values.start, closeTo(50, 0.01));
+
+    // Drag the end thumb towards the center.
+    await tester.pumpAndSettle();
+    final Offset rightTarget = topLeft + (bottomRight - topLeft) * 0.7;
+    await tester.dragFrom(rightTarget, middle - rightTarget);
+    expect(values.end, closeTo(50, 0.01));
+
+    // Drag the end thumb apart.
+    await tester.pumpAndSettle();
+    await tester.dragFrom(middle, (bottomRight - topLeft) * 0.3);
+    expect(values.end, closeTo(80, 0.01));
+  });
+
+  testWidgets('Range Slider thumbs can be dragged together and the end thumb can be dragged apart (discrete RTL)', (WidgetTester tester) async {
+    RangeValues values = RangeValues(30, 70);
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.rtl,
+        child: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return MediaQuery(
+              data: MediaQueryData.fromWindow(window),
+              child: Material(
+                child: Center(
+                  child: RangeSlider(
+                    values: values,
+                    min: 0,
+                    max: 100,
+                    divisions: 10,
+                    onChanged: (RangeValues newValues) {
+                      setState(() {
+                        values = newValues;
+                      });
+                    },
+                  ),
+                ),
+              ),
+            );
+          },
+        ),
+      ),
+    );
+
+    // Get the bounds of the track by shifting by the the overlay radius.
+    final Offset topLeft = tester.getTopLeft(find.byType(RangeSlider)).translate(24, 0);
+    final Offset bottomRight = tester.getBottomRight(find.byType(RangeSlider)).translate(-24, 0);
+    final Offset middle = topLeft + bottomRight / 2;
+
+    // Drag the end thumb towards the center.
+    final Offset leftTarget = topLeft + (bottomRight - topLeft) * 0.3;
+    await tester.dragFrom(leftTarget, middle - leftTarget);
+    expect(values.end, closeTo(50, 0.01));
+
+    // Drag the start thumb towards the center.
+    await tester.pumpAndSettle();
+    final Offset rightTarget = topLeft + (bottomRight - topLeft) * 0.7;
+    await tester.dragFrom(rightTarget, middle - rightTarget);
+    expect(values.start, closeTo(50, 0.01));
+
+    // Drag the end thumb apart.
+    await tester.pumpAndSettle();
+    await tester.dragFrom(middle, -(bottomRight - topLeft) * 0.3);
+    expect(values.end, closeTo(80, 0.01));
+  });
+
+  testWidgets('Range Slider uses the right theme colors for the right shapes', (WidgetTester tester) async {
```

Comment:
This test is quite large, consider splitting it into smaller tests.

Reply:
Good idea. I was able to split this up quite a bit.